services/py-api/src/database/repository/participants_repository.py

```diff
@@ -28,8 +29,14 @@ async def fetch_all(self) -> Result:
 
     async def update(
         self, obj_id: str, input_data: BaseModel, session: Optional[AsyncIOMotorClientSession] = None, **kwargs: Any
-    ) -> Result:
-        raise NotImplementedError()
+    ) -> Result[Ok, Exception]:
+        try:
+            LOG.debug(f"Updating participant with id {obj_id}")
+            await self._collection.find_one_and_update({"_id": ObjectId(obj_id)}, {"$set": input_data}, session=session)
+            return Ok(value=True)
+        except Exception as e:
+            LOG.error(f"Updating participant with id {obj_id} failed due to err {e}")
+            return Err(e)
```

Comment:
Please use [LOG.exception](https://www.structlog.org/en/stable/api.html#structlog.stdlib.BoundLogger.exception) so that we can preserve the stack trace in our log files. 

For more info: https://betterstack.com/community/guides/logging/structlog/#logging-exceptions-with-structlog
https://www.structlog.org/en/stable/api.html#structlog.processors.StackInfoRenderer